src/app/features/errors/app-error-boundary.tsx

```diff
@@ -1,58 +1,103 @@
-import { Box, Stack, styled } from 'leather-styles/jsx';
+import { useRouteError } from 'react-router-dom';
+
+import BroadcastError from '@assets/images/unhappy-face-ui.png';
+import { SharedComponentsSelectors } from '@tests/selectors/shared-component.selectors';
+import { Box, Flex, HStack, styled } from 'leather-styles/jsx';
 
 import { Prism } from '@app/common/clarity-prism';
-import { HasChildren } from '@app/common/has-children';
-import { ErrorBoundary, FallbackProps, useErrorHandler } from '@app/features/errors/error-boundary';
-import { openGithubIssue } from '@app/features/errors/utils';
-import { useErrorStackTraceState } from '@app/store/ui/ui.hooks';
+import { useClipboard } from '@app/common/hooks/use-copy-to-clipboard';
+import { Button } from '@app/ui/components/button/button';
 import { CodeBlock } from '@app/ui/components/codeblock';
-import { Title } from '@app/ui/components/typography/title';
+import { Link } from '@app/ui/components/link/link';
+import { CopyIcon } from '@app/ui/icons';
 
-function ErrorFallback({ error, resetErrorBoundary }: FallbackProps) {
-  const [value] = useErrorStackTraceState();
+import { useToast } from '../toasts/use-toast';
 
-  return (
-    <Stack gap="space.06" flexGrow={1}>
-      <Title>Something went wrong</Title>
-      <Box className="error-codeblock" maxWidth="100vw" overflow="hidden">
-        {value && (
-          <CodeBlock
-            border="default"
-            code={value}
-            flexShrink={1}
-            overflow="auto"
-            language="bash"
-            maxHeight="305px"
-            maxWidth="100%"
-            prism={Prism as any}
-            width="100%"
-          />
-        )}
-      </Box>
-      <Stack mt="auto" gap="space.04">
-        <styled.button onClick={resetErrorBoundary} type="button">
-          Reload extension
-        </styled.button>
-        <styled.button
-          onClick={() => openGithubIssue({ message: error.message, stackTrace: value })}
-          type="button"
-        >
-          Report issue on GitHub
-        </styled.button>
-      </Stack>
-    </Stack>
-  );
-}
+export function RouterErrorBoundary() {
+  const error = useRouteError() as Error;
+  const errorObj = JSON.parse(JSON.stringify(error, Object.getOwnPropertyNames(error)));
```

Comment:
This looks sketchy. Let's do some runtime checking. 

Something like?
`if (!isError) return children;` or nav home?

And why the stringify/parse? I have no idea why this is happening, and nor might the next person reading this code.

Reply:
hm, I added this a while ago, not recall now why. removed

Reply:
not sure if there could be `!isError` as it's error boundary. but even if it could -> better to show at least "Unknown error" msg. 
what if error happened on home screen?